### PR TITLE
Spacecat 634 jh2 vm benchmarktest

### PR DESCRIPTION
**Pull request by fraugabel**

### CODE REVIEW

File: features/openstack_testflow.feature [hunk 1]
```diff
@@ -3,25 +3,25 @@ Feature: Test creation and deletion of resources
...
     Given I connect to OpenStack
 
-    Then I should be able to create <router_quantity> routers
-    Then I should be able to create <network_quantity> networks
-    Then I should be able to create <subnet_quantity> subnets
-    Then I should be able to create <security_group_quantity> security groups
-    Then I should be able to create <security_group_rules_quantity> security group rules
-    Then I create <quantity_volumes> volumes
-
+   # Then I should be able to create <router_quantity> routers
```
Comment (piobig2871) [inline features/openstack_testflow.feature:6]: why is that commented? comment is for all of the comments in the file

File: features/steps/definitions.py [hunk 2]
```diff
@@ -529,20 +529,66 @@ def create_a_jumphost(context, jumphost_name: str, network_name: str, keypair_na
...
+    # @given("I have deployed a JH with the name {jh_name}")
+    # def initialize(context, jh_name: str):
+    @given("I have deployed {jh_quantity:d} JHs")
+    def initialize(context, jh_quantity):
+        # servers=context.client.compute.servers()
+        # lookup= context.test_name+"-jh"
+        # context.jh=[]
+        # jh=None   
+        # for name in servers:
+        #     print(f"server {name.name}")
```
Comment (piobig2871) [inline features/steps/definitions.py:546]: that can't be merged, please uncomment or delete

File: features/steps/definitions.py [hunk 3]
```diff
@@ -529,20 +529,66 @@ def create_a_jumphost(context, jumphost_name: str, network_name: str, keypair_na
...
+        for i in range(1, jh_quantity + 1):
+            context.vm_ip_address=context.jh[i-1]['ip']
+            print(f"ip {1}: {context.vm_ip_address}, {conn_test} ")
+            # try:
+            context.execute_steps('''
+                Then I should be able to SSH into the VM
+                Then I should be able to collect all VM IPs
+                And be able to ping all IPs to test {conn_test} connectivity
+            ''')
+            # except:
```
Comment (piobig2871) [inline features/steps/definitions.py:582]: delete?

File: features/steps/definitions.py [hunk 4]
```diff
@@ -552,19 +598,22 @@ def test_internet_connectivity(context):
...
+        context.ips=tools.collect_ips(context.client)
+    
+    @then("be able to ping all IPs to test {conn_test} connectivity") 
+    def ping_ips_test(context, conn_test: str):
+        tot_ips=len(context.ips)
         for ip in context.ips:
-            result, assertline = context.ssh_client.test_internet_connectivity(ip, tot_ips)
-        print(result)
+            result,assertline=context.ssh_client.test_internet_connectivity(conn_test, ip,tot_ips)
+        print(f"result {result}")
```
Comment (piobig2871) [inline features/steps/definitions.py:608]: change to logger

File: features/steps/definitions.py [hunk 5]
```diff
@@ -552,19 +598,22 @@ def test_internet_connectivity(context):
...
-        context.ips = tools.collect_ips(context.client)
-
-    @then("be able to ping all IPs")
-    def ping_ips_test(context):
-        tot_ips = len(context.ips)
+        context.ips=tools.collect_ips(context.client)
+    
+    @then("be able to ping all IPs to test {conn_test} connectivity") 
+    def ping_ips_test(context, conn_test: str):
+        tot_ips=len(context.ips)
```
Comment (piobig2871) [inline features/steps/definitions.py:605]: fix pep

File: features/steps/definitions.py [hunk 6]
```diff
@@ -552,19 +598,22 @@ def test_internet_connectivity(context):
...
         for ip in context.ips:
-            result, assertline = context.ssh_client.test_internet_connectivity(ip, tot_ips)
-        print(result)
+            result,assertline=context.ssh_client.test_internet_connectivity(conn_test, ip,tot_ips)
+        print(f"result {result}")
+        # if result[1] != 0:
```
Comment (piobig2871) [inline features/steps/definitions.py:609]: why do we need that comment? is it to be deleted or added into code?

File: features/steps/tools.py [hunk 7]
```diff
@@ -175,15 +174,39 @@ def check_volumes_created(client, test_name):
...
-    print("collecting ips")
-    ports = client.network.ports()
     ips = []
-    for port in ports:
-        for fixed_ip in port.fixed_ips:
-            ips.append(fixed_ip["ip_address"])
+    print("collecting ips")
+    floating_ips = client.network.ips()
+    print("floating")
+    for ip in floating_ips:
```
Comment (piobig2871) [inline features/steps/tools.py:181]: remove prints, change to logger

File: features/steps/tools.py [hunk 8]
```diff
@@ -175,15 +174,39 @@ def check_volumes_created(client, test_name):
...
+    # for port in ports:
+    #     for fixed_ip in port.fixed_ips:
+    #         ips.append(fixed_ip['ip_address'])
     return ips
 
 
+def collect_jhs(client, test_name):
+    servers=client.compute.servers()
+    lookup= test_name+"-jh"
+    lookup="default-jh" # just for testing delete
```
Comment (piobig2871) [inline features/steps/tools.py:194]: do you know that it will take only the second variable declaration?

File: libs/ConnectivityClient.py [hunk 9]
```diff
@@ -141,11 +176,33 @@ def create_script(self,ips,c=1,w=3,c_retry=1,w_retry=3):
...
+
+            Args:
+                class
+            Returns:
+                prints current working dir 
+            Raises:
+                Exception:
+        """
         directory = self.execute_command("pwd")
         print(f"Current working directory on server {self.host}: {directory}")
```
Comment (piobig2871) [inline libs/ConnectivityClient.py:207]: use logger or pprint here to format response

File: features/openstack_testflow.feature (hunk 1 above)
Comment (fraugabel) [reply features/openstack_testflow.feature:6]: this was for testing reasons, as you work on the testflow again this is all outdated

File: features/steps/tools.py (hunk 8 above)
Comment (fraugabel) [reply features/steps/tools.py:194]: yes it's for testing purposes, until we have an agreement on the naming conventions of the jh

**Pull request closed without merging**